### PR TITLE
chore: refactor the DeeplinkManager into smaller parts and add unit tests

### PR DESCRIPTION
**Pull request by omridan159**
## **Description**

Refactored the `DeeplinkManager` class into smaller, more manageable components, enhancing code readability and maintainability.
This change simplifies future modifications and debugging. Additionally, comprehensive unit tests were added to ensure each component functions correctly, boosting overall software reliability.

## **Related issues**

Fixes: #

## **Manual testing steps**

IOS
1. Test the deep links behavior with the `devnext` project on a testing device.
2. Test the deep links behavior with the `devreactnative` project on a testing device.

Android:
1. Test the deep links behavior with the `devnext` project on a testing device.
2. Test the deep links behavior with the `devreactnative` project on a testing device.

### **After**

https://github.com/MetaMask/metamask-mobile/assets/61094771/147fc998-cf49-448e-a71b-2af8006c9225

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/core/DeeplinkManager/ParseManager/handleParseWCProtocol.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import WC2Manager from '../../WalletConnect/WalletConnectV2';
+import extractURLParams from './extractURLParams';
+
+export function handleParseWCProtocol({
```
Comment (abretonc7s) [inline app/core/DeeplinkManager/ParseManager/handleParseWCProtocol.ts:4]: that files seems to connect to wc, probably would need a more appropriate name

**Review by abretonc7s — CHANGES_REQUESTED** [review]
I think we should rename all the handleParseMetamask/Universal/Walle by handleMetamask/Univer/Walle.

It would make it easier to match the filename to the functionality without opening the file

**Review by abretonc7s — APPROVED** [review]

**Review by deeeed — APPROVED** [review]
LGTM

**Review by abretonc7s — APPROVED** [review]
LGTM

**Review by abretonc7s — DISMISSED** [review]
LGTM

**Comment by christopherferreira9** [comment]
E2E: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/bfad41f3-66ff-48a9-98c5-9bb80027d9ae

**Comment by christopherferreira9** [comment]
**Checked for both Android and iOS**

- Multiple sign variants
- SwitchChain
- AddChain
- Session persistence
- WalletConnect
- Deeplink variants from https://metamask.github.io/metamask-deeplinks/#

Only one issue found @omridan159 .
Steps to reproduce:
1. Open https://metamask.github.io/metamask-deeplinks/#
2. Select "Payment request"
3. Enter any details you want with and without chainId (on separate tries)
4. Tap the link 
5. See issue

Video:

https://github.com/MetaMask/metamask-mobile/assets/104831203/fc9e2594-9bc5-42d1-a58a-c24e4184e82d

**Review by deeeed — APPROVED** [review]

**Review by andreahaku — APPROVED** [review]
LGTM

**Pull request merged**